connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -756,7 +759,19 @@ private static Object convertToConnect(Schema schema, JsonNode jsonValue) {
     }
 
     private interface LogicalTypeConverter {
-        JsonNode toJson(Schema schema, Object value, JsonConverterConfig config);
+        JsonNode toJson(Schema schema, Object value, CachedConfigs config);
         Object toConnect(Schema schema, JsonNode value);
     }
+
+    private static final class CachedConfigs {
+        final boolean isKey;
+        final boolean enableSchemas;
+        final DecimalFormat decimalFormat;
+
+        private CachedConfigs(final boolean isKey, final boolean enableSchemas, final DecimalFormat decimalFormat) {
```

Comment:
Nit: wouldn't it be easier and perhaps less error-prone to pass the `JsonConverterConfig` into this constructor and set these fields directly from the config? It would also make it easier to call this constructor above, turning the multi-line call into a single line call.

Reply:
+1 I like that idea!